include/c_utilities/types/string_array.h

```diff
@@ -31,14 +31,38 @@ typedef struct C_UTILITIES_PUBLIC_TYPE string_array_t
   char ** data;
 } string_array_t;
 
+/// Returns an empty string array struct
+/*
+ * This function returns an empty and zero initialized string array struct.
+ * Calling utilities_string_array_fini on this instance leads to undefined
+ * behavior.
```

Comment:
The tests seem to confirm my expectation, maybe there is just a typo in the docs here.